pkg/operation/botanist/secrets.go

```diff
@@ -1166,10 +1167,10 @@ func dnsNamesForService(name, namespace string) []string {
 
 func dnsNamesForEtcd(namespace string) []string {
 	names := []string{
-		fmt.Sprintf("%s-0", v1beta1constants.StatefulSetNameETCDMain),
-		fmt.Sprintf("%s-0", v1beta1constants.StatefulSetNameETCDEvents),
+		fmt.Sprintf("%s-local", v1beta1constants.ETCDMain),
```

Comment:
Check for backward compatibility with existing shoots where secrets are already deployed and not reconciled further.

Reply:
Where do you delete these old secrets so that they are reconciled? I don't see the code.